### DISCUSSION TITLE
Noob questions for contributing

### DISCUSSION THREAD

**Discussion by nth-chile** [glushchenko/fsnotes#1214] [Q&A] [answered]
I love the idea behind this app. The iCloud-based file syncing is brilliant, and the local git backup with history is brilliant too. The Bear-style nested tag system, the plaintext feeling of the editor is just what I was looking for.

I am a JavaScript developer. I have written iOS apps using React Native, and I have used Xcode a little. But I know next to nothing about native Apple development. I looked through the FSNotes source code and I think I should be able to grasp it in time.

I cloned the repo and opened it in Xcode, and I have three questions:

1. Why are there so many targets? Could you please explain the difference between:
  - FSNotes
  - FSNotes (iCloud Documents)
  - FSNotes (Notarized)
  - FSNotesCore macOS
  - FSNotesCore macOSTests

2. How do I make sure that my notes and the git backup are safe while I'm working on the app? Can I run it in a sandboxed environment or something?

3. The certs/profiles in Signing & Capabilities are red/have errors. I'm assuming this is expected -- it's because I'm not glushchenko so I shouldn't be able to sign the app. But, is there anything I need to do here, or just ignore those errors while working?

Thanks!

**Comment by glushchenko** [comment, marked as answer]
1. For developers usable only one first FSNotes scheme.

- FSNotes for development without sign
- FSNotes (iCloud Documents) - builds for AppStore
- FSNotes (Notarized) - builds for GitHub
- FSNotesCore - build shared code for iOS and macOS apps

2. You can run it enywhere, backups not deleted in any cases.

3. Just ignore or replace, but not commit it in repo.
